### PR TITLE
Run cbmc c++ regression as part of default regression test

### PR DESCRIPTION
**Pull request by andreast271**
Running cbmc on c++ files was not part of standard regression tests, which lead to a regression. 
This change set fixes the regression, adds a few tests, and adds cbmc c++ tests to the standard regression.

### CODE REVIEW

**Review by reuk — COMMENTED** [review]
Should be enabled in the cmake files too.

**Comment by andreast271** [comment]
There are other test directories in the Makefile that are missing from cmake: cbmc-cover, goto-analyzer-taint, goto-gcc. Should they also be added?

**Comment by reuk** [comment]
In that case, don't worry about it. I'll put together a PR enabling the remaining directories once this has been merged.

**Comment by andreast271** [comment]
I needed to add a commit to this PR anyway to disable the new tests on appveyor. I added the missing test directories to cmake at the same time. Exception: goto-gcc. I wasn't sure about the correct cmake file for it.

**Comment by thk123** [comment]
I think it is not worth at this stage getting these tests to work on Windows (they would need to be wrapped in some kind of script to do the fixing up). Perhaps something like this for now?

```
if(NOT WIN32)
    add_test_pl_tests(
      "$<TARGET_FILE:goto-cc>"
    )
endif()
```

@reuk can say whether this is a good way of doing this.

**Comment by andreast271** [comment]
Some of the goto-gcc tests fail on Travis when running as part of the cmake build. See https://travis-ci.org/diffblue/cbmc/jobs/289820097. 
I will revert the commit that added the goto-gcc tests to the cmake build. Fixing goto-gcc tests under cmake is not the point of the PR.

**Comment by thk123** [comment]
Yup that's reasonable - I've raised an issue #1500 to remind us to fix this, but it certainly shouldn't block this PR.

**Review by thk123 — APPROVED** [review]

File: appveyor.yml
Comment (tautschnig) [inline appveyor.yml:86]: Would this work if we had #1260 merged?

Comment (andreast271) [reply appveyor.yml:86]: Yes. You should be able to merge this PR and #1260 in any order. I'd recommend merging them close to each other.

Comment (tautschnig) [reply appveyor.yml:86]: How about merging this one first, and then reverting this particular change in #1260?

Comment (andreast271) [reply appveyor.yml:86]: Sorry, I misunderstood. The cbmc-cpp tests never ran on any platform. This PR enables them in general, and then disables them specifically for windows. They don't work on windows irrespective of #1260, since they #include <assert.h>.

**Review by tautschnig — APPROVED** [review]

**Pull request merged**
